docs/pages/guides/spend-permissions/api-reference.mdx

```diff
@@ -0,0 +1,275 @@
+| `end`       | `uint48` | Timestamp this spend permission is valid until (unix seconds).              |
+| `salt`      | `uint256`| An arbitrary salt to differentiate unique spend permissions with otherwise identical data. |
+| `extraData` | `bytes`  | Arbitrary data to include in the signature.                                 |
+
+---
+
+#### `SpendPermissionBatch`
+
+| Field         | Type                   | Description                                                                 |
+|---------------|------------------------|-----------------------------------------------------------------------------|
+| `account`     | `address`              | Smart account this spend permission is valid for.                           |
+| `period`      | `uint48`               | Time duration for resetting used allowance on a recurring basis (seconds).  |
+| `start`       | `uint48`               | Timestamp this spend permission is valid after (unix seconds).              |
+| `end`         | `uint48`               | Timestamp this spend permission is valid until (unix seconds).              |
+| `permissions` | `PermissionDetails[]`  | Array of PermissionDetails structs defining properties that apply per-permission. |
+
+---
+
+#### `PermissionDetails`
+
+| Field       | Type     | Description                                                                 |
+|-------------|----------|-----------------------------------------------------------------------------|
+| `spender`   | `address`| Entity that can spend user funds.                                           |
+| `token`     | `address`| Token address (ERC-7528 ether address or ERC-20 contract).                  |
+| `allowance` | `uint160`| Maximum allowed value to spend within a recurring period.                   |
+| `salt`      | `uint256`| An arbitrary salt to differentiate unique spend permissions with otherwise identical data. |
+| `extraData` | `bytes`  | Arbitrary data to include in the signature.                                 |
+
+---
+
+#### `PeriodSpend`
+
+| Field       | Type     | Description                                                                 |
+|-------------|----------|-----------------------------------------------------------------------------|
+| `start`     | `uint48` | Start time of the period (unix seconds).                                    |
+| `end`       | `uint48` | End time of the period (unix seconds).                                      |
+| `spend`     | `uint160`| Accumulated spend amount for period.                                        |
+
+---
+
+### Contract functions
+#### `approve`
+
+Approve a spend permission via a direct call from the account. Only callable by the account owner specified in the spend permission.
+
+```solidity
+function approve(SpendPermission calldata spendPermission) external requireSender(spendPermission.account);
```

Comment:
Would remove the `requireSender` modifier for cleanliness and your comment explains the behavior already which is good